### ISSUE TITLE
[QA] Testing using Postman for login

### ISSUE THREAD

**Issue by luzelenatt** [unmsmfisi-socialapplication/social_app#321]
Consider test scenarios

**Comment by PaulaRV01** [comment]
According to Login Acceptance Test Cases #320 @luzelenatt, testing was performed using postman. The results are as follows:

<html>
<body>
<google-sheets-html-origin>

**ID CP** | **Scenario** |  **Condición de ejecución** | **JSON** | **Expected Result** | **Result Obtained**
-- | -- | -- | -- | -- | --
CP_01 | Inicio de Sesión Exitoso | El usuario debe estar registrado en el aplicativo social. |  {    "username": "my_user",    "password": "hopefully this is not a plain text password"} | {     "response": "Authentication successful",     "status": "OK" } | {  "response": "Authentication successful",  "status": "OK" }
CP_02 | Contraseña Incorrecta | El usuario debe estar registrado en el aplicativo social. | {    "username": "my_user",    "password": "hopefully this is not a plain text"} | { "response": "Invalid password", "status": "NOPASSWORD"} | {    "response": "Invalid credentials",    "status": "BADCREDENTIALS"}
CP_03 | Usuario no Encontrado | El usuario debe estar registrado en el aplicativo social. |  {    "username": "nonexistent_user",    "password": "hopefully this is not a plain text password"} | {     "response": "User not found",     "status": "NOTFOUND" } | {     "response": "User not found",     "status": "NOTFOUND" }
CP_04 | Campos no Completados | El usuario debe estar registrado en el aplicativo social. |  {"username": "", "password": ""} | {     "response": "User not found",     "status": "NOTFOUND" } | {    "response": "User not found",    "status": "NOTFOUND"}
CP_05 | Campos no Completados | El usuario debe estar registrado en el aplicativo social. |  {    "username": "my_user",    "password": ""} | {    "response": "Invalid credentials",    "status": "BADCREDENTIALS"} | {    "response": "Invalid credentials",    "status": "BADCREDENTIALS"}
CP_06 | Campos no Completados | El usuario debe estar registrado en el aplicativo social. |  {    "username": "",    "password": "hopefully this is not a plain text password"} | {    "response": "User not found",    "status": "NOTFOUND"} | {    "response": "User not found",    "status": "NOTFOUND"}
CP_07 | Campos Completados con Error | El usuario debe estar registrado en el aplicativo social. |  {    "username": "invalid_user",    "password": "invalid_password"} | {     "response": "User not found",     "status": "NOTFOUND" } | {     "response": "User not found",     "status": "NOTFOUND" }


</body>
</html>

**Comment by julianntorres** [comment]
> According to Login Acceptance Test Cases #320 @luzelenatt, testing was performed using postman. The results are as follows:
> 
> * Considering business rules [[UX/UI] Create bussiness rules #243](https://github.com/unmsmfisi-socialapplication/social_app/issues/243), both the username and password must be alphanumeric for their authenticity. According to this: test cases 1,2,3,5,6 are erroneous due to these considerations. Therefore, backend must take this into consideration.
> * In test cases 5 and 6, validation is verified with missing password and user data respectively. Therefore the expected results would be: NOT PASSWORD and NOT USER.
> * While test case 2 is tested with an invalid password. Therefore, the expected result would be BADCREDENTIALS
> 
> **ID CP**	**Scenario**	**Condición de ejecución**	**JSON**	**Expected Result**	**Result Obtained**
> CP_01	Inicio de Sesión Exitoso	El usuario debe estar registrado en el aplicativo social.	{ "username": "myuser12", "password": "Social@12"}	{ "response": "Authentication successful", "status": "OK" }	{ "response": "User not found", "status": "NOTFOUND" }
> CP_02	Contraseña Incorrecta	El usuario debe estar registrado en el aplicativo social.	{ "username": "myuser12", "password": "hopefullythisisnotaplaintext"}	{ "response": "Invalid credentials", "status": "BADCREDENTIALS"}	{ "response": "User not found", "status": "NOTFOUND" }
> CP_03	Usuario no Encontrado	El usuario debe estar registrado en el aplicativo social.	{ "username": "nonexistentuser", "password": "Social@12"}	{ "response": "User not found", "status": "NOTFOUND" }	{ "response": "User not found", "status": "NOTFOUND" }
> CP_04	Campos no Completados	El usuario debe estar registrado en el aplicativo social.	{"username": "", "password": ""}	{ "response": "User not found", "status": "NOTFOUND" }	{ "response": "User not found", "status": "NOTFOUND" }
> CP_05	Campos no Completados	El usuario debe estar registrado en el aplicativo social.	{ "username": "myuser12", "password": ""}	{ "response": "not password", "status": "NOPASSWORD"}	{ "response": "User not found", "status": "NOTFOUND" }
> CP_06	Campos no Completados	El usuario debe estar registrado en el aplicativo social.	 { "username": "", "password": "Social@12"}	{ "response": "not user", "status": "NOTUSER"}	{ "response": "User not found", "status": "NOTFOUND" }
> CP_07	Campos Completados con Error	El usuario debe estar registrado en el aplicativo social.	{ "username": "invaliduser", "password": "invalidpassword"}	{ "response": "User not found", "status": "NOTFOUND" }	{ "response": "User not found", "status": "NOTFOUND" }
> For more detail, review the document [LOGIN_ACCEPTATION TEST CASES_POSTMAN](https://docs.google.com/spreadsheets/d/1NfdZW-MB-r7clxOG5MxbdCQklpK9e3tuBd-k9Rdm9cw/edit#gid=928425818)

Dear QA Team,

Thank you for your detailed observations on the login component test cases.

- Regarding test cases 1, 2, and 3: The discrepancies in the expected results for these tests stem from the hardcoded user in the handler. This user was explicitly set by the professor days prior to this feedback. The test cases you have executed use a different user than the one hardcoded, leading to the inconsistencies. I'd like to point out that this is not a logical error within the component but rather an alignment issue between the hardcoded user in the code (given the absence of a database connection at this point) and the user you've recently introduced. We will update the hardcoded user to the one you mentioned to ensure the desired response is achieved.

- Regarding test case 5: The expected result you've provided,

`{
 "response": "not password",
 "status": "NOPASSWORD"
}
`

Will never be achieved as it's not defined in the endpoint's documentation. There is no such response or status within the actual code. I assume this expected result was drafted without consulting the component's documentation or code.

- Regarding test case 6: Similarly, the expected result,
`{
    "response": "not user",
    "status": "NOTUSER"
}
`

Will never be met. It seems to be another instance where the expectation is not in alignment with the component's design or documentation.


To summarize, there seems to be a misalignment between QA's expectations and the current structure and documentation of the login component. Nevertheless, for the benefit of the project, we will adjust the hardcoded user in the handler for test cases 1, 2, and 3. However, it's crucial to acknowledge that test cases 5 and 6 are invalid as they don't align with the logic or documentation of the component.

The login component documentation it's deeply detailed here : [click](https://github.com/unmsmfisi-socialapplication/social_app/pull/76)

Best regards,

Julian.

**Comment by PaulaRV01** [comment]
According to the documentation provided by Backend, the respective update was made for CP 5 and CP 6, the expected response to
{ "response": "User not found", "status": "NOTFOUND" } and { "response": "Invalid password", "status": "NOPASSWORD"} respectively

However, it is important to mention that errors will occur if both the username and password are not considered alphanumeric.

The results are:
CP 3,4,6 and 7 are correct. While CP 1,2 and 5 are incorrect due to the consideration of alphanumeric username and password.

We await the prompt update for the execution of the Test Cases.

<html>
<body>
<google-sheets-html-origin>

**ID CP** | **Scenario** |  **Condición de ejecución** | **JSON** | **Expected Result** | **Result Obtained**
-- | -- | -- | -- | -- | --
CP_01 | Inicio de Sesión Exitoso | El usuario debe estar registrado en el aplicativo social. |  {    "username": "myuser12",    "password": "Social@12"} | {     "response": "Authentication successful",     "status": "OK" } | { "response": "User not found", "status": "NOTFOUND" }
CP_02 | Contraseña Incorrecta | El usuario debe estar registrado en el aplicativo social. | {    "username": "myuser12",    "password": "hopefullythisisnotaplaintext"} | { "response": "Invalid password", "status": "NOPASSWORD"} | { "response": "User not found", "status": "NOTFOUND" }
CP_03 | Usuario no Encontrado | El usuario debe estar registrado en el aplicativo social. |  {    "username": "nonexistentuser",    "password": "Social@12"} | {     "response": "User not found",     "status": "NOTFOUND" } | { "response": "User not found", "status": "NOTFOUND" }
CP_04 | Campos no Completados | El usuario debe estar registrado en el aplicativo social. |  {"username": "", "password": ""} | {     "response": "User not found",     "status": "NOTFOUND" } | { "response": "User not found", "status": "NOTFOUND" }
CP_05 | Campos no Completados | El usuario debe estar registrado en el aplicativo social. |  {    "username": "myuser12",    "password": ""} | { "response": "Invalid password", "status": "NOPASSWORD"} | { "response": "User not found", "status": "NOTFOUND" }
CP_06 | Campos no Completados | El usuario debe estar registrado en el aplicativo social. |  {    "username": "",    "password": "Social@12"} | { "response": "User not found", "status": "NOTFOUND" } | { "response": "User not found", "status": "NOTFOUND" }
CP_07 | Campos Completados con Error | El usuario debe estar registrado en el aplicativo social. |  {    "username": "invaliduser",    "password": "invalidpassword"} | {     "response": "User not found",     "status": "NOTFOUND" } | { "response": "User not found", "status": "NOTFOUND" }


</body>
</html>

For more detail, review the document [LOGIN_ACCEPTATION TEST CASES_POSTMAN](https://docs.google.com/spreadsheets/d/1NfdZW-MB-r7clxOG5MxbdCQklpK9e3tuBd-k9Rdm9cw/edit#gid=928425818)

**Comment by julianntorres** [comment]
> According to the documentation provided by Backend #76 , the respective update was made for CP 5 and CP 6, the expected response to `{ "response": "Invalid password", "status": "NOPASSWORD"} ` and `{ "response": "User not found", "status": "NOTFOUND" }` respectively
> 
> However, it is important to mention that errors will occur if both the username and password are not considered alphanumeric.
> 
> The results are: CP 3,4,6 and 7 are correct. While CP 1,2 and 5 are incorrect due to the consideration of alphanumeric username and password.
> 
> We await the prompt update.
> 
> **ID CP**	**Scenario**	**Condición de ejecución**	**JSON**	**Expected Result**	**Result Obtained**
> CP_01	Inicio de Sesión Exitoso	El usuario debe estar registrado en el aplicativo social.	{ "username": "myuser12", "password": "Social@12"}	{ "response": "Authentication successful", "status": "OK" }	{ "response": "User not found", "status": "NOTFOUND" }
> CP_02	Contraseña Incorrecta	El usuario debe estar registrado en el aplicativo social.	{ "username": "myuser12", "password": "hopefullythisisnotaplaintext"}	{ "response": "Invalid password", "status": "NOPASSWORD"}	{ "response": "User not found", "status": "NOTFOUND" }
> CP_03	Usuario no Encontrado	El usuario debe estar registrado en el aplicativo social.	{ "username": "nonexistentuser", "password": "Social@12"}	{ "response": "User not found", "status": "NOTFOUND" }	{ "response": "User not found", "status": "NOTFOUND" }
> CP_04	Campos no Completados	El usuario debe estar registrado en el aplicativo social.	{"username": "", "password": ""}	{ "response": "User not found", "status": "NOTFOUND" }	{ "response": "User not found", "status": "NOTFOUND" }
> CP_05	Campos no Completados	El usuario debe estar registrado en el aplicativo social.	{ "username": "myuser12", "password": ""}	{ "response": "Invalid password", "status": "NOPASSWORD"}	{ "response": "User not found", "status": "NOTFOUND" }
> CP_06	Campos no Completados	El usuario debe estar registrado en el aplicativo social.	 { "username": "", "password": "Social@12"}	{ "response": "User not found", "status": "NOTFOUND" }	{ "response": "User not found", "status": "NOTFOUND" }
> CP_07	Campos Completados con Error	El usuario debe estar registrado en el aplicativo social.	{ "username": "invaliduser", "password": "invalidpassword"}	{ "response": "User not found", "status": "NOTFOUND" }	{ "response": "User not found", "status": "NOTFOUND" }
> For more detail, review the document [LOGIN_ACCEPTATION TEST CASES_POSTMAN](https://docs.google.com/spreadsheets/d/1NfdZW-MB-r7clxOG5MxbdCQklpK9e3tuBd-k9Rdm9cw/edit#gid=928425818)

To inform you that the sample response has been updated in alignment with the endpoint documentation to prevent any further confusion. Regarding the alphanumeric consideration for username and password, I'd like to clarify that such a validation does not impact the login process. The responsibility for ensuring a username and password are alphanumeric lies with the registration process, not the login. The login simply authenticates and compares the data sent from the frontend with existing data.

It's important to note that tests CP1, CP2, and CP5 are providing the correct response, which is "User Not Found". This is due to the hardcoded user, as set by the professor, not being "myuser12", but rather "my_user", as he modified it last Friday. You can see it on [this commit](https://github.com/unmsmfisi-socialapplication/social_app/commit/4edcdb853ba828bf854355a3fbbb8320e76e0acd)

I appreciate your understanding and am here to address any further concerns or questions.

**Comment by luzelenatt** [comment]
As part of the QA team, @PaulaRV01 and @luzelenatt, we have reviewed the [update](https://github.com/unmsmfisi-socialapplication/social_app/commit/2ca576e2cc8b36fa7916c05237ba8b43263d478f) made by the Backend team, which was requested in the meeting last Sunday 8th, in which the leaders of the UX/UI and frontend teams participated.

In addition, we evaluated a [small modification](https://github.com/unmsmfisi-socialapplication/social_app/pull/76) in the backend documentation, which we consider valid and which provides a better understanding in the JSON response when a user incorrectly enters their password.

With the collaboration of DevOps for the integration of these new changes in the [deployment](https://github.com/orgs/unmsmfisi-socialapplication/projects/1/views/1?filterQuery=%5Bdevops%5D&pane=issue&itemId=37651991), we will proceed to test again for each of the scenarios.

**Comment by PaulaRV01** [comment]
### **Test Case: TC01 - Inicio de Sesión Exitoso**
we have the next json code:
`{
    "username": "myuser12",
    "password": "Social@12"
}`
the results in postman show us:
![Image](https://github.com/unmsmfisi-socialapplication/social_app/assets/88557553/5f6932d8-02f1-4d1c-847a-15b93458e880)
it has been succesfuly proved

### **Test Case: TC02 - Contraseña Incorrecta**
we have the next json code:
`{
    "username": "myuser12",
    "password": "hopefullythisisnotaplaintext"
}`
the results in postman show us:
![Image](https://github.com/unmsmfisi-socialapplication/social_app/assets/88557553/ecd50eea-5b63-493a-926f-5eca1e3e6158)
it has been succesfuly proved

### **Test Case: TC03 - Usuario no Encontrado**
we have the next json code:
`{
    "username": "nonexistentuser",
    "password": "Social@12"
}`
the results in postman show us:
![Image](https://github.com/unmsmfisi-socialapplication/social_app/assets/88557553/0423db4e-02ff-47bc-943b-6b501ebb1afa)
it has been succesfuly proved

### **Test Case: TC04 - Campos no Completados**
we have the next json code:
`{"username": "", "password": ""}`
the results in postman show us:
![Image](https://github.com/unmsmfisi-socialapplication/social_app/assets/88557553/825ed9db-5f31-49e7-9bc3-dc4476069620)
it has been succesfuly proved

### **Test Case: TC05 - Password no Completado**
we have the next json code:
`{
    "username": "myuser12",
    "password": ""
}`
the results in postman show us:
![Image](https://github.com/unmsmfisi-socialapplication/social_app/assets/88557553/267d445c-aeee-4a6f-b7e9-3135f567875d)
it has been succesfuly proved

### **Test Case: TC06 - Usuario no Completado**
we have the next json code:
`{
    "username": "",
    "password": "Social@12"
}`
the results in postman show us:
![Image](https://github.com/unmsmfisi-socialapplication/social_app/assets/88557553/1982a288-5eee-4b7a-96a4-eac7b53241b1)
it has been succesfuly proved


### **Test Case: TC07 - Campos Completados con Error**
we have the next json code:
`{
    "username": "invaliduser",
    "password": "invalidpassword"
}`
the results in postman show us:
![Image](https://github.com/unmsmfisi-socialapplication/social_app/assets/88557553/8ae54095-cfcb-4be9-ac4c-07bce8aaedb0)
it has been succesfuly proved

**Comment by eds000n** [comment]
let's add this test so its executed by the CI everytime we go from `develop` to `main`, please talk with @unmsmfisi-socialapplication/devops

**Comment by luzelenatt** [comment]
Ok @eds000n , I'm going to contact the Devops representative to look into this issue.
